hs_core/models.py

```diff
@@ -1158,12 +1158,17 @@ class MyResourceContentType(pages.Page, hs_core.AbstractResource):
 
     @classmethod
     def bag_url(cls, resource_id):
+        from hs_core.hydroshare.utils import get_resource_by_shortkey
         bagit_path = getattr(settings, 'IRODS_BAGIT_PATH', 'bags')
         bagit_postfix = getattr(settings, 'IRODS_BAGIT_POSTFIX', 'zip')
-
         bag_path = "{path}/{resource_id}.{postfix}".format(path=bagit_path,
                                                            resource_id=resource_id,
                                                            postfix=bagit_postfix)
+        res = get_resource_by_shortkey(resource_id)
+        if res.resource_federation_path:
+            rel_fed_path = res.resource_federation_path[1:]
+            bag_path = os.path.join(rel_fed_path, bag_path)
+
         istorage = IrodsStorage()
```

Comment:
Don't we need to use the IrodsStorage('federated') when the if condition is true?